### PR TITLE
Fix StackOverFlow when execute /give all

### PR DESCRIPTION
**Pull request by iTruth**
## Description

Please carefully read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md) before making any pull requests.
And, **Do not make a pull request to merge into stable unless it is a hotfix. Use the development branch instead.**
## Issues fixed by this PR
* ItemUseGainFlycloak and ItemUseGainCostume will cause StackOverFlow
* putItem might throws exception to terminate /give all

## Type of changes

- [x] Bug fix
- [ ] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseGainCostume.java [hunk 1]
```diff
@@ -14,7 +17,10 @@ public ItemUseGainCostume(String[] useParam) {
 
     @Override
     public boolean useItem(UseItemParams params) {
-        params.player.getInventory().addItem(this.i);  // TODO: Currently this returns false for all virtual items - need to have a proper success/fail
+        AvatarCostumeData costumeData = GameData.getAvatarCostumeDataItemIdMap().get(this.i);
+        if (costumeData != null && !params.player.getCostumeList().contains(costumeData.getId())) {
+            params.player.addCostume(costumeData.getId());
+        }
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseGainCostume.java]:
```suggestion
        if (GameData.getAvatarCostumeDataMap().containsKey(this.i)) {
            params.player.addCostume(this.i);
        }
```

File: src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseGainFlycloak.java [hunk 2]
```diff
@@ -14,7 +17,10 @@ public ItemUseGainFlycloak(String[] useParam) {
 
     @Override
     public boolean useItem(UseItemParams params) {
-        params.player.getInventory().addItem(this.i);  // TODO: Currently this returns false for all virtual items - need to have a proper success/fail
+        AvatarFlycloakData flycloakData = GameData.getAvatarFlycloakDataMap().get(this.i);
+        if (flycloakData != null && !params.player.getFlyCloakList().contains(this.i)) {
+            params.player.addFlycloak(this.i);
+        }
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseGainFlycloak.java]: `addFlycloak` is a set operation, it doesn't have ill effects if the set already contains the flycloak.
```suggestion
        if (GameData.getAvatarFlycloakDataMap().containsKey(this.i)) {
            params.player.addFlycloak(this.i);
        }
```

File: src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseGainFlycloak.java [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 import emu.grasscutter.game.props.ItemUseOp;
 
+import emu.grasscutter.data.GameData;
+import emu.grasscutter.data.excels.AvatarFlycloakData;
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseGainFlycloak.java]:
```suggestion
```

File: src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseGainCostume.java [hunk 4]
```diff
@@ -2,6 +2,9 @@
 
 import emu.grasscutter.game.props.ItemUseOp;
 
+import emu.grasscutter.data.GameData;
+import emu.grasscutter.data.excels.AvatarCostumeData;
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseGainCostume.java]:
```suggestion
```

**Review by Birdulon — CHANGES_REQUESTED** [review]

**Comment by iTruth** [comment]
@Birdulon you're right, thanks for reminding

**Review by Birdulon — APPROVED** [review]

**Pull request merged**